src/bp-xprofile/bp-xprofile-filters.php

```diff
@@ -1061,3 +1061,187 @@ function bp_xprofile_exclude_display_name_profile_fields( $args ){
 
 	return $args;
 }
+
+/**
+ * Add xprofile notification repair list item.
+ *
+ * @param array $repair_list Repair list items.
+ *
+ * @return array Repair list items.
+ *
+ * @since BuddyBoss 1.6.2
+ */
+function bb_xprofile_repeater_field_repair( $repair_list ) {
+	$get_bp_xprofile_migration = (bool) bp_get_option( 'bp_xprofile_migration' );
+	if ( false === $get_bp_xprofile_migration ) {
+		$repair_list[] = array(
+			'bp-xprofile-repeater-field-repair',
+			__( 'Repair xprofile repeater field repeated.', 'buddyboss' ),
+			'bb_xprofile_repeater_field_repair_callback',
+		);
+	}
+	return $repair_list;
+}
+
+/**
+ * This function will work as migration process which will remove duplicate repeater field from database by repair tool.
+ * Also remove remove unnecessary data that may have remained after the migration process.
+ * This function will be called only once.
+ *
+ * @uses bb_xprofile_repeater_field_migration
+ *
+ * @since BuddyBoss 1.6.2
+ */
+function bb_xprofile_repeater_field_repair_callback() {
+	$offset = filter_input( INPUT_POST, 'offset', FILTER_VALIDATE_INT );
+	if ( 1 === $offset ) {
+		$offset = 0;
+	} else {
+		$offset = $offset;
+	}
+	// Function will do migrate code for the repeated fields.
+	return bb_xprofile_repeater_field_migration( $offset, $callback = true );
+}
+
+/**
+ * This function will work as migration process which will remove duplicate repeater field from database.
+ * Also remove remove unnecessary data that may have remained after the migration process.
+ * This function will be called only once.
+ *
+ * @param int     $offset offset.
+ * @param boolean $callback callback.
+ *
+ * @return array
+ *
+ * @since BuddyBoss 1.6.2
+ */
+function bb_xprofile_repeater_field_migration( $offset, $callback ) {
```

Comment:
@sadrul I have checked the code and it's not working for me. Can you please test and update accordingly?